File: Backend/Services/geminiService.js
// Thin wrapper around the Gemini REST API.
//
// Uses the built-in global `fetch` (Node 18+, which Express 5 already
// requires) instead of an SDK, so there's nothing extra to install.

const MODEL_NAME = 'gemini-2.5-flash';
const ENDPOINT = `https://generativelanguage.googleapis.com/v1beta/models/${MODEL_NAME}:generateContent`;

async function generateContent(prompt) {
  const apiKey = process.env.GEMINI_API_KEY;

  if (!apiKey || apiKey === 'your_gemini_api_key_here') {
    return {
      success: false,
      content: null,
      error:
        'GEMINI_API_KEY is not set. Add a real key to Backend/.env to enable AI features.',
    };
  }

  try {
    const response = await fetch(`${ENDPOINT}?key=${apiKey}`, {
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify({
        contents: [{ parts: [{ text: prompt }] }],
      }),
    });

    const data = await response.json();

    if (!response.ok) {
      return {
        success: false,
        content: null,
        error: data?.error?.message ?? 'Gemini request failed.',
      };
    }

    const text =
      data?.candidates?.[0]?.content?.parts
        ?.map((part) => part.text ?? '')
        .join('') ?? '';

    if (!text) {
      return {
        success: false,
        content: null,
        error: 'Gemini returned an empty response.',
      };
    }

    return { success: true, content: text, error: null };
  } catch (err) {
    return { success: false, content: null, error: err.message };
  }
}

module.exports = { generateContent, MODEL_NAME };

File: Backend/Services/GeminiService.py
from google import genai
from google.genai import types

from Config import Config


class GeminiService:

    def __init__(self):

        self.client = genai.Client(
            api_key=Config.GEMINI_API_KEY
        )

        self.modelName = "gemini-2.5-flash"

    def GenerateContent(self, prompt):


        try:

            response = self.client.models.generate_content(
                model=self.modelName,
                contents=prompt
            )

            return {
                "success": True,
                "content": response.text,
                "error": None
            }

        except Exception as error:

            return {
                "success": False,
                "content": None,
                "error": str(error)
            }

    def GenerateResume(self, prompt):


        return self.GenerateContent(prompt)

    def GenerateCoverLetter(self, prompt):


        return self.GenerateContent(prompt)

    def RewriteDocument(self, prompt):

        return self.GenerateContent(prompt)

    def TestConnection(self):


        try:

            response = self.client.models.generate_content(
                model=self.modelName,
                contents="Reply with the word Connected."
            )

            return {
                "success": True,
                "message": response.text
            }

        except Exception as error:

            return {
                "success": False,
                "message": str(error)
            }